### PR TITLE
Slovenian translation, by Jernej Omulec

### PR DESCRIPTION
**Pull request by pedrosino**

### CODE REVIEW

**Comment by pedrosino** [comment]
@viroulep help?

**Comment by viroulep** [comment]
Oops, we don't check if the locale is supported by `TwitterCldr` before creating the collator :s 
cc @jfly for information.

Turns out Slovenian is not in the supported locales, therefore it fails. One cool solution would be to add it following the steps [here](https://github.com/twitter/twitter-cldr-rb#adding-new-locales), but I feel we should handle the non-supported locale case anyway.
I've fixed it like [this](https://github.com/viroulep/worldcubeassociation.org/commit/c713ee0340535f68159dfdda006df8088dd319af) locally, feel free to cherry-pick the commit/do the same fix.

Also the bootstrap views Slovenian locale is almost empty [here](https://github.com/hisea/devise-bootstrap-views/blob/4175430cd32c41f4634bcb740bfdfbc59bfafee6/locales/sl.yml) (including the "sign in/up" :( ), I expect Jernej to come back to us saying some part of the website, so it may be a good idea to warn him about that first, and suggest him to contribute directly to the `devise-bootstrap-views` gem.

File: WcaOnRails/config/locales/locales.rb [hunk 1]
```diff
@@ -65,5 +65,9 @@ module Locales
       "flag_id": "hr",
       "name": "Hrvatski",
     },
+    "sl": {
+      "flag_id": "sl",
```
Comment (viroulep) [inline WcaOnRails/config/locales/locales.rb]: Flag id for Slovenian is `si`(!).

**Comment by jfly** [comment]
> but I feel we should handle the non-supported locale case anyway.

Why do you feel that way? I think it would be better for us to fail tests and fix this somehow*. Otherwise how would we notice this?

That's frustrating that other code has left empty strings for Slovenian in their codebase. @Luis-J-Ianez would be happy if we explicitly defined those strings in our `en.yml`, because he's not happy with their Spanish translations. Maybe we can kill two birds with one stone here?

*Either by adding support for that language to Twitter CLDR, or by could see explicitly adding a mapping for the problematic language, as we already do for `pt-BR`, `zh-CN`, and `zh-TW` (see https://github.com/thewca/worldcubeassociation.org/blob/9c957a4e1cd48d05b546449f3b03e6a6b50941b0/WcaOnRails/app/models/country.rb#L55-L59).

**Comment by viroulep** [comment]
> Otherwise how would we notice this?

Right, somehow I wanted to avoid the runtime error, but if it happens in tests it's actually fine.
The short term solution would be to map `sl` to something else (`en` seems to work good, they use almost the same alphabet as Croatian, so maybe `hr` would be more accurate).
It would be a decent solution until someone (we?) add it to TwitterCLDR.

> That's frustrating that other code has left empty strings for Slovenian in their codebase. @Luis-J-Ianez would be happy if we explicitly defined those strings in our en.yml, because he's not happy with their Spanish translations. Maybe we can kill two birds with one stone here?

Ideally we would submit such changes directly to `devise-bootstrap-views`, but if we cannot complete or update the locale upstream, a workaround may be to add a custom devise locale file for some locales (to override them out of our main locale file, so that not everyone has to override them).

**Comment by jfly** [comment]
I'd be okay with defining them in our en.yml and just requiring all translators to do them. It's probably not that much work, and would probably result in fewer surprises for them (they're not going to expect to see text on the website that they didn't write). Furthermore, if we break this out into separate files, we can't use @jonatanklosko's lovely app.

**Comment by pedrosino** [comment]
How should I proceed here?

**Comment by pedrosino** [comment]
Argh, I'm making such a mess. I don't know what this Twitter stuff is about and probably need some sleep.

**Comment by viroulep** [comment]
We use a part of `TwitterCLDR` to be able to sort countries/continents correctly whatever the language is.

@jfly is right here about how to fix the issue I think: we should simply put: `'sl': 'en',` below [here](https://github.com/thewca/worldcubeassociation.org/pull/1321/files#diff-2c543e1650c254ecc83a33005c769c7fR58), and forget the part where we check if the locale is supported (as @jfly said, it's ok since it will crash during tests).

For the rest of the untranslated part of the website (ie: devise), I think we should invite Jernej to complete the locale upstream (he even can use Jonatan's app for that, by specifying the file he wants to translate, and setting the existing Slovenian file as a reference).
(But we don't need to wait for this to merge this PR)

**Comment by pedrosino** [comment]
Shouldn't it be 'hr' instead of 'en', as per your comment before?

**Comment by viroulep** [comment]
I'd say so, but as I said I'm just guessing: I'm not fluent in either Slovenian or Croatian :p
I'm sending a mail to Jernej now.

File: WcaOnRails/app/models/country.rb [hunk 2]
```diff
@@ -56,7 +56,9 @@ def self.localized_sort_by!(wca_locale, array, &block)
       'pt-BR': 'pt',
       'zh-CN': 'zh',
       'zh-TW': 'zh-Hant',
+      'sl': 'hr',
```
Comment (jfly) [inline WcaOnRails/app/models/country.rb]: This looks great, but could we make an issue for this and add a link to it here?

**Pull request merged**